rudder-rest/src/main/scala/com/normation/rudder/rest/ApiAuthorization.scala

```diff
@@ -103,33 +106,59 @@ class AclApiAuthorization(logger: Log, userService: UserService, aclEnabled: ()
       else None
     }
 
+    def checkACL(acl: List[ApiAclElement], path: ApiPath, action: HttpAction): Option[String] = {
+      if(AclCheck(acl, path, endpoint.schema.action)) {
+        Some("ok")
+      } else {
+        None
+      }
+    }
+
     val user = userService.getCurrentUser
     for {
       // we want to compare the exact path asked by the user to take care of cases where he only has
       // access to a limited subset of named resourced for the endpoint.
       path <- requestPath.drop(endpoint.prefix).leftMap(msg => ApiError.BadRequest(msg, endpoint.schema.name))
-      ok   <- (user.getApiAutz match {
-                case ApiAuthz.None     =>
-                  logger.debug(s"User '${user.actor.name}' does not have any authorizations.")
+      ok   <- ((aclEnabled(), user.getApiAuthz, user.account) match {
+                /*
+                 * We need to check the account type. It is ok for a user account to have that
+                 * kind of ACL rights, because it's the way we use to map actual user account role to internal API
+                 * access (for ex: an user with role "node r/w/e" can change node properties, which use "update setting"
+                 * API, so he must have access to that, but not to other configuration API).
+                 * On the other hand, if we are dealing with an actual API account, we need to check its kind more
+                 * preciselly:
+                 * - a system account keep ACL eval (most likelly equiv to RW)
+                 * - a standard account get RO (we can broke third party app behovior but don't open too big security
+                 *   hole with updates)
+                 * - an user API account is disabled.
+                 */
+                // without plugin, api account linked to user are disabled
+                case (false, _, RudderAccount.Api(ApiAccount(_, ApiAccountKind.User, _, _, _, _, _, _))) =>
+                  logger.warn(s"API account linked to a user account '${user.actor.name}' is disabled because the API Authorization plugin is disabled.")
+                  None //token link to user account is a plugin only feature
+
+                // without plugin but ACL configured, standard api account are change to RO to avoid unwanted mod
+                case (false, ApiAuthz.ACL(acl), RudderAccount.Api(ApiAccount(_, _:ApiAccountKind.PublicApi, _, _, _, _, _, _))) =>
+                  logger.info(s"API account '${user.actor.name}' has ACL authorization but no plugin allows to interpret them. Revert to R0 rights.")
+                  checkACL(acl.filter( _.actions == HttpAction.GET), path, endpoint.schema.action)
```

Comment:
So we will filter acl to allow only authorized get from acl list ? I thought we would fall back to checkRO so acl feature was only available to plugin owners ?

Reply:
hum, I thought I didn't commited it, not sure what happened. I'm changing it back.